blueprint/listener.go

```diff
@@ -0,0 +1,111 @@
+package blueprint
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/micromdm/mdm"
+	"github.com/micromdm/micromdm/checkin"
+	"github.com/micromdm/micromdm/command"
+	"github.com/micromdm/micromdm/profile"
+	"github.com/micromdm/micromdm/pubsub"
+	"github.com/pkg/errors"
+)
+
+func (db *DB) ApplyToDevice(ctx context.Context, svc command.Service, bp *Blueprint, udid string) error {
+
+	var requests []*mdm.CommandRequest
+	for _, appURL := range bp.ApplicationURLs {
+		requests = append(requests, &mdm.CommandRequest{
+			RequestType: "InstallApplication",
+			UDID:        udid,
+			InstallApplication: mdm.InstallApplication{
+				ManifestURL:     appURL,
+				ManagementFlags: 1,
+			},
+		})
+	}
+
+	for _, p := range bp.ProfileIdentifiers {
+		foundProfile, err := db.profDB.ProfileById(p)
+		if err != nil {
+			if profile.IsNotFound(err) {
+				fmt.Printf("Profile ID %s in Blueprint %s does not exist\n", p, bp.Name)
+				continue
+			}
+			fmt.Println(errors.Wrap(err, "fetching profile"))
```

Comment:
use `fmt.Printf` for printing instead of errors.Wrap. The `Wrap` method is only useful to be able to inspect an error from a higher level, but you're not returning here.